### PR TITLE
Prevent reloading already loaded Katana plugin modules.

### PR DESCRIPTION
**Pull request by arthurp**

### CODE REVIEW

File: python/katana/plugin.py [hunk 1]
```diff
@@ -56,13 +56,13 @@ def load_module(self, fullname):
...
-        module = importlib.import_module(self._insert_table[fullname])
+        aliased_name = self._insert_table[fullname]
         if fullname in sys.modules:
-            raise ImportError(
-                f"module {fullname} already exists while trying to insert "
-                f"{self._insert_table[fullname]} at that name."
-            )
+            module = sys.modules[aliased_name]
+        else:
+            module = importlib.import_module(aliased_name)
```
Comment (gliga) [inline python/katana/plugin.py:63]: should aliasd_name,fullname be inserted here in _insert_table? (or that is done elsewhere)

Comment (gliga) [reply python/katana/plugin.py:63]: ok based on that assert above, it is probably inserted somewhere else

Comment (arthurp) [reply python/katana/plugin.py:63]: Yea the table is build in `_register_plugin_loader` and passed to the constructor.

File: python/katana/plugin.py [hunk 2]
```diff
@@ -56,13 +56,13 @@ def load_module(self, fullname):
 
         assert fullname in self._insert_table
 
-        # This will get the module from sys.modules if it is already loaded.
-        module = importlib.import_module(self._insert_table[fullname])
+        aliased_name = self._insert_table[fullname]
         if fullname in sys.modules:
```
Comment (gliga) [inline python/katana/plugin.py]: if fullname was in sys.modules we already returned; aliased_name?

Comment (arthurp) [reply python/katana/plugin.py]: This code creates an alias of the module named `aliased_name` at the name `fullname`. Does that clear it up? I'm not really sure what you are trying to ask.

Comment (gliga) [reply python/katana/plugin.py]: there is an if statement at the beginning of the method that checks if fullname in sys.modules. When does fullname change between the two if statements?

Comment (arthurp) [reply python/katana/plugin.py]: Ah. I see. You are right. It should be `aliased_name` there. I will fix it. I have no idea how this worked then. Hmm. That's worrisome.

File: python/katana/plugin.py [hunk 3]
```diff
@@ -56,13 +56,13 @@ def load_module(self, fullname):
...
         assert fullname in self._insert_table
 
-        # This will get the module from sys.modules if it is already loaded.
-        module = importlib.import_module(self._insert_table[fullname])
+        aliased_name = self._insert_table[fullname]
+        if aliased_name in sys.modules:
+            module = sys.modules[aliased_name]
+        else:
+            module = importlib.import_module(aliased_name)
         if fullname in sys.modules:
```
Comment (gliga) [inline python/katana/plugin.py]: should this statement be only under "else"?

Comment (arthurp) [reply python/katana/plugin.py]: It can be. I'll move it.

**Review by gliga — APPROVED** [review]

**Pull request merged**